src/eval/interpreter.zig

```diff
@@ -1370,9 +1421,125 @@ pub const Interpreter = struct {
         bool_ptr.* = result_val;
     }
 
+    /// Handle boolean scalar tag expressions (True/False) by setting the appropriate boolean value
+    fn handleBooleanScalarTag(self: *Interpreter, result_value: StackValue, tag_name: Ident.Idx) EvalError!void {
+        const tag_text = self.env.idents.getText(tag_name);
+        const bool_value: u8 = if (std.mem.eql(u8, tag_text, "True")) 1 else if (std.mem.eql(u8, tag_text, "False")) 0 else {
+            self.traceError("Invalid boolean tag: {s}", .{tag_text});
+            return error.InvalidBooleanTag;
+        };
+
+        const bool_ptr = @as(*u8, @ptrCast(@alignCast(result_value.ptr.?)));
+        bool_ptr.* = bool_value;
+
+        self.traceInfo("Boolean scalar tag: {s} = {}", .{ tag_text, bool_value });
+    }
+
+    /// Get the return layout of a closure by resolving its lambda expression's type
+    fn getClosureReturnLayout(self: *Interpreter, closure: *const Closure) EvalError!Layout {
+        // Get the type Var for the lambda expression
+        const lambda_var = ModuleEnv.varFrom(closure.lambda_expr_idx);
+
+        // Resolve the lambda's type to get the function type
+        const lambda_resolved = self.env.types.resolveVar(lambda_var);
+
+        // Extract the return type from the function
+        switch (lambda_resolved.desc.content) {
+            .structure => |structure| switch (structure) {
+                .fn_pure => |func| {
+                    // Ensure the return type variable is fully resolved before getting layout
+                    const ret_resolved = self.env.types.resolveVar(func.ret);
+
+                    // Check if it's still unresolved (flex_var/rigid_var)
+                    switch (ret_resolved.desc.content) {
+                        .flex_var, .rigid_var => {
+                            self.traceInfo("LAYOUT SYSTEM NOT READY: Lambda return type is still unresolved ({}).", .{ret_resolved.desc.content});
+                            self.traceInfo("This means the type system hasn't finished constraint solving before layout computation.", .{});
+                            self.traceInfo("The type system shows the correct type in TYPES section, but the type variables", .{});
+                            self.traceInfo("passed to layout computation are still flex_var/rigid_var instead of concrete types.", .{});
+                            self.traceInfo("This will be fixed when the type system implementation is completed.", .{});
+                            return error.NotImplemented;
```

Comment:
@jaredramirez can you confirm this is the correct approach here. Should the `Func` struct `ret` field be fully resolved to the lamda return type? Should we expect that to have propagated all the way through and a flex or rigid var probably means that some feature in type checking just isn't implemented yet?